### PR TITLE
Fix a couple of 404's in release-team/README

### PR DESCRIPTION
**Pull request by gtirloni**

### CODE REVIEW

**Comment by nikopen** [comment]
thank you!

/lgtm
/cc @justaugustus

**Comment by nikhita** [comment]
/assign @justaugustus 
for approval

File: release-team/README.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ is undefined.
 
 ### During "Security" releases
 For all unplanned or embargoed releases
-- Facilitate security releases following the under the [Security Release Process](security-release-process-documentation/security-release-process.md)
+- Facilitate security releases following the under the [Security Release Process](../security-release-process-documentation/security-release-process.md)
```
Comment (justaugustus) [inline release-team/README.md]:
```suggestion
- Facilitate security releases following the under the [Security Release Process](/security-release-process-documentation/security-release-process.md)
```

File: release-team/README.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ there are 3 types of Kubernetes releases:
 | Test Infra | [Test Infra Handbook](role-handbooks/test-infra/README.md) |
 | Bug Triage | [Bug Triage Handbook](role-handbooks/bug-triage/README.md) |
 | Branch Manager | [Branch Manager Handbook](role-handbooks/branch-manager/README.md) |
-| Docs | [Doc Handbook](role-handbooks/docs/README.md) |
+| Docs | [Doc Handbook](role-handbooks/documentation-guides) |
```
Comment (justaugustus) [inline release-team/README.md]: Please remove this change.
We should fix the broken link by conforming the Docs handbook to the other role handbooks.

**Comment by justaugustus** [comment]
Held until my review is addressed.
/hold

**Pull request closed without merging**